pkg/deployment/resources/gateway_config.go

```diff
@@ -0,0 +1,295 @@
+//
+// DISCLAIMER
+//
+// Copyright 2024 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+
+package resources
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strconv"
+	"time"
+
+	bootstrapAPI "github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v3"
+	clusterAPI "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	coreAPI "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	endpointAPI "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	listenerAPI "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	routeAPI "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	routerAPI "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/router/v3"
+	httpConnectionManagerAPI "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"sigs.k8s.io/yaml"
+
+	shared "github.com/arangodb/kube-arangodb/pkg/apis/shared"
+)
+
+type Redirect KV[string, []string]
+
+type KV[K comparable, V any] struct {
+	K K
+	V V
+}
+
+func RenderGatewayConfigYAML(dbServiceAddress string, redirects ...Redirect) ([]byte, error) {
```

Comment:
Remove redirects - not needed in the first version|

Otherwise move KV under pkg/util/dict.go